app/search/search-form.tsx

```diff
@@ -13,7 +13,9 @@ export default function SearchForm({
 
   React.useEffect(() => {
     if (inputRef.current) {
-      inputRef.current.value = initialSearchTerm ?? '';
+      if (!inputRef.current.value && initialSearchTerm) {
+        inputRef.current.value = initialSearchTerm;
+      }
```

Comment:
This change preserves the current search term input value during an HMR event, while maintaining the original goal of ensuring the search term matches what's in the URL (if there is one).